`SanitizerCommon-ubsan-*-Linux :: sanitizer_coverage_*.cpp` test failures (glibc 2.39?)

Three new failures while testing HEAD as of 60a904b2ad9842b93cc5fa0ad5bda5e22c550b7e. I'm guessing this is because I've upgraded my system to glibc 2.39.

```
FAIL: SanitizerCommon-ubsan-i386-Linux :: sanitizer_coverage_symbolize.cpp (5102 of 7007)
******************** TEST 'SanitizerCommon-ubsan-i386-Linux :: sanitizer_coverage_symbolize.cpp' FAILED ********************
Exit Code: 1

Command Output (stderr):
--
RUN: at line 6: DIR=/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-i386-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp_workdir
+ DIR=/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-i386-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp_workdir
RUN: at line 7: rm -rf $DIR
+ rm -rf /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-i386-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp_workdir
RUN: at line 8: mkdir -p $DIR
+ mkdir -p /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-i386-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp_workdir
RUN: at line 9: cd $DIR
+ cd /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-i386-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp_workdir
RUN: at line 10: /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/lib/llvm/19/bin/clang  --driver-mode=g++ -gline-tables-only -fsanitize=undefined  -m32 -funwind-tables  -I/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test -ldl -O0 -fsanitize-coverage=trace-pc-guard /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp -o /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-i386-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp
+ /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/lib/llvm/19/bin/clang --driver-mode=g++ -gline-tables-only -fsanitize=undefined -m32 -funwind-tables -I/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test -ldl -O0 -fsanitize-coverage=trace-pc-guard /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp -o /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-i386-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp
RUN: at line 11: env UBSAN_OPTIONS=coverage=1 /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-i386-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp 2>&1 | FileCheck /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp
+ env UBSAN_OPTIONS=coverage=1 /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-i386-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp
+ FileCheck /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp
/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp:28:11: error: CHECK: expected string not found in input
// CHECK: SanitizerCoverage: ./sanitizer_coverage_symbolize.{{.*}}.sancov: 2 PCs written
          ^
<stdin>:1:5: note: scanning from here
main
    ^
<stdin>:4:1: note: possible intended match here
SanitizerCoverage: ./sanitizer_coverage_symbolize.cpp.tmp.248595.sancov: 4 PCs written
^

Input file: <stdin>
Check file: /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1: main 
check:28'0         X error: no match found
            2: foo 
check:28'0     ~~~~
            3: foo 
check:28'0     ~~~~
            4: SanitizerCoverage: ./sanitizer_coverage_symbolize.cpp.tmp.248595.sancov: 4 PCs written 
check:28'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:28'1     ?                                                                                       possible intended match
>>>>>>

--

********************
FAIL: SanitizerCommon-ubsan-x86_64-Linux :: sanitizer_coverage_inline8bit_counter_default_impl.cpp (5323 of 7007)
******************** TEST 'SanitizerCommon-ubsan-x86_64-Linux :: sanitizer_coverage_inline8bit_counter_default_impl.cpp' FAILED ********************
Exit Code: 1

Command Output (stderr):
--
RUN: at line 6: /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/lib/llvm/19/bin/clang  --driver-mode=g++ -gline-tables-only -fsanitize=undefined  -m64 -funwind-tables  -I/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test -ldl -O0 /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_inline8bit_counter_default_impl.cpp -fsanitize-coverage=inline-8bit-counters,pc-table -o /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp
+ /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/lib/llvm/19/bin/clang --driver-mode=g++ -gline-tables-only -fsanitize=undefined -m64 -funwind-tables -I/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test -ldl -O0 /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_inline8bit_counter_default_impl.cpp -fsanitize-coverage=inline-8bit-counters,pc-table -o /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp
RUN: at line 7: rm -f /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp-counters /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp-pcs
+ rm -f /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp-counters /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp-pcs
RUN: at line 8: env UBSAN_OPTIONS="cov_8bit_counters_out=/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp-counters cov_pcs_out=/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp-pcs verbosity=1"  /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp 2>&1 | FileCheck /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_inline8bit_counter_default_impl.cpp
+ env 'UBSAN_OPTIONS=cov_8bit_counters_out=/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp-counters cov_pcs_out=/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp-pcs verbosity=1' /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp
+ FileCheck /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_inline8bit_counter_default_impl.cpp
RUN: at line 11: wc -c /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp-counters | grep "^2 "
+ wc -c /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_inline8bit_counter_default_impl.cpp.tmp-counters
+ grep '^2 '

--

********************
FAIL: SanitizerCommon-ubsan-x86_64-Linux :: sanitizer_coverage_symbolize.cpp (5327 of 7007)
******************** TEST 'SanitizerCommon-ubsan-x86_64-Linux :: sanitizer_coverage_symbolize.cpp' FAILED ********************
Exit Code: 1

Command Output (stderr):
--
RUN: at line 6: DIR=/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp_workdir
+ DIR=/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp_workdir
RUN: at line 7: rm -rf $DIR
+ rm -rf /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp_workdir
RUN: at line 8: mkdir -p $DIR
+ mkdir -p /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp_workdir
RUN: at line 9: cd $DIR
+ cd /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp_workdir
RUN: at line 10: /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/lib/llvm/19/bin/clang  --driver-mode=g++ -gline-tables-only -fsanitize=undefined  -m64 -funwind-tables  -I/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test -ldl -O0 -fsanitize-coverage=trace-pc-guard /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp -o /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp
+ /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/lib/llvm/19/bin/clang --driver-mode=g++ -gline-tables-only -fsanitize=undefined -m64 -funwind-tables -I/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test -ldl -O0 -fsanitize-coverage=trace-pc-guard /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp -o /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp
RUN: at line 11: env UBSAN_OPTIONS=coverage=1 /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp 2>&1 | FileCheck /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp
+ FileCheck /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp
+ env UBSAN_OPTIONS=coverage=1 /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt_build/test/sanitizer_common/ubsan-x86_64-Linux/Output/sanitizer_coverage_symbolize.cpp.tmp
/var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp:28:11: error: CHECK: expected string not found in input
// CHECK: SanitizerCoverage: ./sanitizer_coverage_symbolize.{{.*}}.sancov: 2 PCs written
          ^
<stdin>:1:5: note: scanning from here
main
    ^
<stdin>:4:1: note: possible intended match here
SanitizerCoverage: ./sanitizer_coverage_symbolize.cpp.tmp.253095.sancov: 4 PCs written
^

Input file: <stdin>
Check file: /var/tmp/portage/sys-libs/compiler-rt-sanitizers-19.0.0_pre20240224/work/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1: main 
check:28'0         X error: no match found
            2: foo 
check:28'0     ~~~~
            3: foo 
check:28'0     ~~~~
            4: SanitizerCoverage: ./sanitizer_coverage_symbolize.cpp.tmp.253095.sancov: 4 PCs written 
check:28'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:28'1     ?                                                                                       possible intended match
>>>>>>

--

********************
```

CC @MaskRay

--- a/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_inline8bit_counter_default_impl.cpp
+++ b/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_inline8bit_counter_default_impl.cpp
@@ -3,7 +3,9 @@
 
 // REQUIRES: has_sancovcc,stable-runtime,linux,x86_64-target-arch
 
-// RUN: %clangxx -O0 %s -fsanitize-coverage=inline-8bit-counters,pc-table -o %t
+/// In glibc 2.39+, fprintf has a nonnull attribute. Disable nonnull-attribute,
+/// which would increase counters for ubsan.
+// RUN: %clangxx -O0 %s -fsanitize-coverage=inline-8bit-counters,pc-table -fno-sanitize=nonnull-attribute -o %t
 // RUN: rm -f %t-counters %t-pcs
 // RUN: env %tool_options="cov_8bit_counters_out=%t-counters cov_pcs_out=%t-pcs verbosity=1" %run %t 2>&1 | FileCheck %s
 

--- a/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp
+++ b/compiler-rt/test/sanitizer_common/TestCases/sanitizer_coverage_symbolize.cpp
@@ -7,7 +7,9 @@
 // RUN: rm -rf $DIR
 // RUN: mkdir -p $DIR
 // RUN: cd $DIR
-// RUN: %clangxx -O0 -fsanitize-coverage=trace-pc-guard %s -o %t
+/// In glibc 2.39+, fprintf has a nonnull attribute. Disable nonnull-attribute,
+/// which would increase counters for ubsan.
+// RUN: %clangxx -O0 -fsanitize-coverage=trace-pc-guard -fno-sanitize=nonnull-attribute %s -o %t
 // RUN: %env_tool_opts=coverage=1 %t 2>&1 | FileCheck %s
 // RUN: rm -rf $DIR
 

